packages/postgres-database/tests/test_utils_aiopg_orm.py

```diff
@@ -0,0 +1,127 @@
+# pylint: disable=no-value-for-parameter
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+
+
+from datetime import datetime
+from typing import Iterator, List
+
+import pytest
+from aiopg.sa.connection import SAConnection
+from aiopg.sa.engine import Engine
+from simcore_postgres_database.models.scicrunch_resources import scicrunch_resources
+from simcore_postgres_database.utils_aiopg_orm import BaseOrm
+
+
+@pytest.fixture
+async def fake_scicrunch_ids(pg_engine: Engine) -> List[str]:
+    row1 = dict(rrid="RRID:foo", name="foo", description="fooing")
+    row2 = dict(rrid="RRID:bar", name="bar", description="barring")
+
+    row_ids = []
+    async with pg_engine.acquire() as conn:
+        for row in (row1, row2):
+            row_id = await conn.scalar(
+                scicrunch_resources.insert()
+                .values(**row)
+                .returning(scicrunch_resources.c.rrid)
+            )
+            assert row_id, f"{row} failed"
+            row_ids.append(row_id)
+
+    return row_ids
+
+
+@pytest.fixture()
+async def scicrunch_orm(pg_engine: Engine) -> Iterator[BaseOrm[str]]:
+    # This is a table without dependencies and therefore easy to use as fixture
+    class ScicrunchOrm(BaseOrm[str]):
+        def __init__(self, connection: SAConnection):
+            super().__init__(
+                scicrunch_resources,
+                connection,
+                readonly={"creation_date", "last_change_date"},
+            )
+
+    async with pg_engine.acquire() as conn:
+        orm_obj = ScicrunchOrm(conn)
+        yield orm_obj
+
+
+async def test_orm_fetch(scicrunch_orm: BaseOrm[str], fake_scicrunch_ids: List[str]):
+
+    # insert 1 and 2
+    scicrunch_id1, scicrunch_id2 = fake_scicrunch_ids
+    assert scicrunch_id1 is not None
+    assert scicrunch_id1 == "RRID:foo"
+    assert scicrunch_id2 is not None
+    assert scicrunch_id2 == "RRID:bar"
+
+    # fetch
+    all_scicrunch_resources = await scicrunch_orm.fetchall()
+    assert len(all_scicrunch_resources) == 2
+
+    # pin-row
+    scicrunch_orm.pin(scicrunch_id1)
+    scicrunch_resource = await scicrunch_orm.fetch()
+    assert scicrunch_resource
+    assert scicrunch_resource.rrid == scicrunch_id1
+
+    # overrides pin-row in a call
+    scicrunch_resource = await scicrunch_orm.fetch(rowid=scicrunch_id2)
+    assert scicrunch_resource
+    assert scicrunch_resource.rrid == scicrunch_id2
+    assert scicrunch_resource in all_scicrunch_resources
+
+    scicrunch_resource = await scicrunch_orm.fetch()
+    assert scicrunch_resource
+    assert scicrunch_resource.rrid == scicrunch_id1
+
+    # partial columns fetch
+    scicrunch_resource = await scicrunch_orm.fetch("name description")
```

Comment:
yeah, not a fan of the string version.

Reply:
haha,... I chose because it is the one you need to write the least (used as well in e.g. ``collections.namedtuples``)

Otherwise it is very easy to change. See the alternatives

```python
scicrunch_resource = await scicrunch_orm.fetch("name description")
# OR
scicrunch_resource = await scicrunch_orm.fetch("name", "description")
# OR
scicrunch_resource = await scicrunch_orm.fetch(["name", "description"])
```

Reply:
I added options one and three above

Reply:
I prefer the third one.

Reply:
could you also add a test the one you've added?